packages/target-tools/src/transforms/transformUtils.js

```diff
@@ -37,9 +41,37 @@ const METRIC_ACTION_PROPERTY_MAP = [
 const METRIC_OPTION_TYPES = values(MetricType);
 export const isMetricOptionType = value => includes(value, METRIC_OPTION_TYPES);
 
-// eslint-disable-next-line no-unused-vars
+function getStepId(stepName) {
+  if (stepName === Step.ENTRY_STEP) {
+    return 0;
+  }
+
+  if (stepName === Step.CONVERSION_STEP) {
+    return 32767;
+  }
+
+  if (stepName === Step.DISPLAY_STEP) {
+    return 2;
+  }
+
+  // campaign step id?
```

Comment:
Have you reproduced the token encoding logic from M2 edge here? Is there something unclear about step Id 1?

Reply:
Hey @XDex , i'm glad you asked.  Yes i reproduced the token encoding logic from M2, but it wasn't clear to me what the "campaign step id" was.  Because it isn't part of the enum, so I couldn't tell when the step ID would be `1`... since the step names that I saw coming from conductor were only `entry`, `display` and `conversion`.  So the logic here is if none of those then return `1` (campaign step id).

Reply:
Got it, thanks for clarifying